packages/salesforcedx-vscode-core/src/commands/util/sfdxCommandlet.ts

```diff
@@ -28,8 +28,22 @@ import { taskViewService } from '../../statuses';
 import { telemetryService } from '../../telemetry';
 import { getRootWorkspacePath } from '../../util';
 
+export enum CommandVersion {
+  Beta = 'beta',
+  Legacy = 'legacy',
+  Default = Beta
+}
+
 export interface FlagParameter<T> {
```

Comment:
nit: I know this isn't new, but since you're here the generic seems unnecessary.  Should we just make flag a `string`.